scipy/optimize/tests/test_minpack.py

```diff
@@ -804,6 +804,27 @@ def test_maxfev_and_bounds(self):
         assert_allclose(popt1, 2, atol=1e-14)
         assert_allclose(popt2, 2, atol=1e-14)
 
+    def test_curvefit_omitnan(self):
+        def exponential(x, a, b):
+            return b * np.exp(a * x)
+
+        np.random.seed(0)
```

Comment:
We should avoid setting the global random state and instead create and use a generator 
```suggestion
        rng = np.random.default_rng(578285731148908)
```

Reply:
Thank you for pointing this out. I didn't know this.